src/App.tsx

```diff
@@ -1,14 +1,46 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useMemo } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
+import { useAppDispatch, useAppSelector } from './app/hooks';
+import { getTodos } from './api';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { actions as todosActions } from './features/todos';
 
 export const App: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const todos = useAppSelector(state => state.todos);
+  const { query, status } = useAppSelector(state => state.filter);
+  const selectedTodo = useAppSelector(state => state.currentTodo);
+
+  useEffect(() => {
+    const fetchTodos = async () => {
+      const todosFromServer = await getTodos();
+
+      dispatch(todosActions.setTodos(todosFromServer));
+    };
+
+    fetchTodos();
+  }, []);
+
+  const lowerQuery = query.toLowerCase();
```

Comment:
You use it only once. It's not necessary to create a separate variable for it. But it's up to you